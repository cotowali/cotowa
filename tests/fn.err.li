fn f1(i int, i int) {}
fn f2(i foo) {}

fn g(i int) string {
  yield '0'
  return i
}
fn g(i int) {}

fn sequence(n int) ...string {
  for v in seq(n) {
    yield v
  }
}

var v = 0
v()

0()
