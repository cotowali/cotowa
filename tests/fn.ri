fn git() string
fn add(a int, b int) int {
  return a + b
}

