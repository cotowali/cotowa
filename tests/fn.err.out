fn.err.li 1,14: `i` is duplicated
    1 | fn f1(i int, i int) {}
      |              ^
fn.err.li 2,9: unknown type `foo`
    2 | fn f2(i foo) {}
      |         ^^^
fn.err.li 5,3: cannot use yield in function that return non-sequence type
    5 |   yield '0'
      |   ^^^^^^^^^
fn.err.li 6,10: mismatched types: `string` (expected) and `int` (found)
    6 |   return i
      |          ^
fn.err.li 8,4: `g` is duplicated
    8 | fn g(i int) {}
      |    ^
fn.err.li 12,11: mismatched types: `string` (expected) and `int` (found)
   12 |     yield v
      |           ^
fn.err.li 17,1: `v` is not function (`int`)
   17 | v()
      | ^^^
fn.err.li 19,1: cannot call `int`
   19 | 0()
      | ^^^
